src/hnsw.c

```diff
@@ -99,7 +99,10 @@ hnswcostestimate(PlannerInfo *root, IndexPath *path, double loop_count,
 {
 	GenericCosts costs;
 	int			m;
-	int			entryLevel;
+	double		entryLevel;
+	double		layer0TuplesMax;
```

Comment:
I think the code will be cleaner if an `int` is used for `entryLevel` and `layer0TuplesMax`, but I can do that in a follow-up commit if that's easier.